common/setups/rasr/config/am_config.py

```diff
@@ -0,0 +1,77 @@
+__all__ = ["TransitionTdp", "SilenceTdp", "AmRasrConfig"]
+
+from dataclasses import dataclass
+from typing import Optional, Tuple, Union
+
+from sisyphus import tk
+
+import i6_core.am as am
+import i6_core.rasr as rasr
+
+
+@dataclass()
+class _Tdp:
+    loop: Union[float, str]
+    forward: Union[float, str]
+    skip: Union[float, str]
+    exit: Union[float, str]
+
+    def get_tuple(
+        self,
+    ) -> Tuple[
+        Union[float, str], Union[float, str], Union[float, str], Union[float, str]
+    ]:
+        return self.loop, self.forward, self.skip, self.exit
+
+
+@dataclass()
+class TransitionTdp(_Tdp):
+    loop: Union[float, str] = 3.0
+    forward: Union[float, str] = 0.0
+    skip: Union[float, str] = "infinity"
+    exit: Union[float, str] = 0.0
+
+
+@dataclass()
+class SilenceTdp(_Tdp):
+    loop: Union[float, str] = 0.0
+    forward: Union[float, str] = 3.0
+    skip: Union[float, str] = "infinity"
+    exit: Union[float, str] = 20.0
+
+
+@dataclass()
+class AmRasrConfig:
+    state_tying: str = "monophone"
+    states_per_phone: int = 3
+    state_repetitions: int = 1
+    across_word_model: bool = True
+    early_recombination: bool = False
+    tdp_scale: float = 1.0
+    tdp_transition: TransitionTdp = TransitionTdp()
+    tdp_silence: SilenceTdp = SilenceTdp()
```

Comment:
```suggestion
    tdp_transition: _Tdp = TransitionTdp()
    tdp_silence: _Tdp = SilenceTdp()
```
if any TDP is allowed here, use base class. Then also rename base class to be public

Reply:
I am wondering how data classes and imutabillity behaves. e.g.
```
a = AmRasrConfig()
a.tdp_transition.loop = 10
b = AmRasrConfig()
print(b.tdp_transition.loop)  # prints 10 
```
ok, the way it is implemented here, all AmRasrConfig share the same `TransitionTdp` instance. 
remedy: use https://docs.python.org/3/library/dataclasses.html#dataclasses.field
```suggestion
    tdp_transition: _Tdp = field(default_factory=TransitionTdp)
    tdp_silence: _Tdp = field(default_factory=SilenceTdp)
```
add `from dataclasses import field` to imports